tests/Constraint/ConstraintTest.php

```diff
@@ -494,6 +500,7 @@ public function testCompile($requireOperator, $requireVersion, $provideOperator,
 
         // Asserts Compiled version returns the same result than standard
         $this->assertSame($m = $require->matches($provide), $this->matchCompiled($require, $provideOperator, $provideVersion));
+        $this->assertSame($m, Intervals::compactConstraint($require)->matches(Intervals::compactConstraint($provide)));
```

Comment:
since the function is about testing compiled constraints, maybe compile the compacted constraint here?

Reply:
Yeah true, will fix